### PR TITLE
[RW-6285] VPC-SC changes to support genomics cohort extraction

### PR DESCRIPTION
**Pull request by calbach**
[Design doc](https://docs.google.com/document/d/1IeHVZt7N2q_kMDFhUAYBOLFzf8fHNu8LgjSGtPIVUAg/edit?ts=5ff5ef06#). The end goal here is to support dynamic genomic cohort extraction implemented on the backend by Cromwell.

- Add BigQuery job user and read session roles, to allow the extraction SA to take BigQuery actions which are billed against researcher projects.
- Add the RW extraction pet SA to the admin / access level, in order to permit backend Cromwell execution from outside the perimeter (Cromwell does not support execution within VPC-SC currently).
- The new extraction SA is added in test/dev only. Other environments will follow later.

### CODE REVIEW

File: profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/perimeters.tf [hunk 1]
```diff
@@ -30,6 +30,25 @@ resource "google_folder_iam_binding" "admin-bucket-writer" {
...
+  folder = google_folder.folder[each.key].name
+  role = "roles/bigquery.jobUser"
+  members = each.value.members
+}
+
+resource "google_folder_iam_binding" "admin-bigquery-read-session-user" {
+  for_each = var.folder_admins
+
+  folder = google_folder.folder[each.key].name
+  role = "roles/bigquery.readSessionUser"
```
Comment (ericsong) [inline profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/perimeters.tf:48]: I've been granting `roles/bigquery.user` when I was doing my tests. We may be able to cut down on the granted permissions by combining some of the less permissive ones but I think we're still missing a few here. At the very least, I think we're missing `bigquery.datasets.create` and `bigquery.datasets.get`.

File: terra-dev.json [hunk 2]
```diff
@@ -91,12 +91,14 @@
             ],
             "access_member_whitelist": [
               "serviceAccount:all-of-us-workbench-test@appspot.gserviceaccount.com",
+              "serviceAccount:pet-118217329794842274136@aou-wgs-cohort-extraction.iam.gserviceaccount.com",
```
Comment (ericsong) [inline terra-dev.json:94]: what's the difference between `terra_dev_aou_test_2` and `terra_dev_aou_test`? I was using the latter.

Comment (calbach) [reply terra-dev.json:94]: _2 is controlled tier

File: profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/perimeters.tf (hunk 1 above)
Comment (calbach) [reply profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/perimeters.tf:48]: This is granting permission on the researcher projects. We shouldn't need to get or create BigQuery datasets within those projects - we just need to be able to run queries / extractions that are billed under them.

I started with bigquery.user, but realized it allows creating datasets, which is excessive privilege here

Comment (ericsong) [reply profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/perimeters.tf:48]: Ah, ok. I was thinking the temp tables would be created in the researcher project but I see they're supposed to stay within VPC-SC project.

**Review by ericsong — APPROVED** [review]

**Comment by calbach** [comment]
Hi @gmalkov , please let me know if you need more information on this or if you want to meet to go over this. I expect most of the changes when applying this to dev, but **some** changes in prod (additional folder IAM roles for our existing service accounts).

**Review by gmalkov — APPROVED** [review]
Applied

**Pull request merged**